Automattic/jetpack
# Pay with Paypal block: Fixed inconsistent currency formatting

Move the siteLocale registration inside Jetpack_Gutenberg, right after the 'jetpack-blocks-editor' is registered, since it's been already used by two blocks: Opentable and Pay with Paypal.

siteLocale property was not properly set in the Customizer because the wp_add_inline_script() call was execute before the "jetpack-blocks-editor" script was registered. When the script is registering, all the associated scripts were unregistered manually. Because of this, the currency format generated in the Widgets was different from the one generated in Customizer, thus triggering a block invalidation issue.





Fixes https://github.com/Automattic/wp-calypso/issues/61483

#### Changes proposed in this Pull Request:

* Move the inline script to Jetpack_Gutenberg, just after the parent script is registered - since siteLocale property is used by both Opentable and Pay with Paypal blocks.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* WPCOM: Apply D76551-code on your sandbox
* Atomic: Use Jetpack Beta on your Atomic site to switch to this branch.
* Make sure that your site uses a non-FSE theme.
* Go to Appearance > Widgets and add a Pay with Paypal block
* Go to Appearance > Customizer > Widgets
* Notice that the block is working properly and not displaying an "Invalid state" error.
* Go to your-site.com/wp-admin/post-new.php and create a new Pay with Paypal block
* Save the post and refresh the page. Make sure that the block remains in a valid state.
